<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Differential Abundance Analysis of Bacteria</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html"></a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Project Overview</a>
</li>
<li>
  <a href="bacteria_community_analysis.html">Bacteria Analysis</a>
</li>
<li>
  <a href="bacteria_differential_abundance_analysis.html">Bacteria Differential Abundance</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Differential Abundance Analysis of Bacteria</h1>

</div>


<script>
$(document).ready(function() {
  $items = $('div#TOC li');
  $items.each(function(idx) {
    num_ul = $(this).parentsUntil('#TOC').length;
    $(this).css({'text-indent': num_ul * 10, 'padding-left': 0});
  });
});
</script>
<pre class="r"><code>phylo.data &lt;- readRDS(&quot;data/bacteria/germs_otus/bacteria_phyloseq.rds&quot;)
phylo.data &lt;- prune_samples(sample_sums(phylo.data)&gt;=200, phylo.data)
phylo.data &lt;- filter_taxa(phylo.data, function(x) sum(x) &gt;= 5, T)</code></pre>
<div id="alpha-diversity" class="section level2">
<h2>Alpha Diversity</h2>
<p>Calculated using Simpsonâ€™s index <span class="math inline">\(H = \sum_{i=1}^{R}p_{i}^{2})\)</span>, it is seen that the treatments allowing more dispersal show greater diversity (increased diverstiy being values less than <span class="math inline">\(1\)</span>).</p>
<pre class="r"><code>richness &lt;- plot_richness(phylo.data, measures = &quot;Simpson&quot;)$data
p &lt;- ggplot(richness, aes(x = Treatment, y = value), color = Treatment)
  p &lt;- p + geom_boxplot(aes(color = Treatment), position = &quot;dodge&quot;)  
  p &lt;- p + ggtitle(&quot;Alpha-Diversity of Treatments&quot;)
  p &lt;- p + ylab(&quot;Shannon Alpha-Diversity&quot;)
  p &lt;- p + theme(plot.title = element_text(hjust = 0.5),
                 axis.title.x = element_blank(),
                 legend.position=&quot;none&quot;)
p</code></pre>
<p><img src="bacteria_differential_abundance_analysis_files/figure-html/alpha-diversity-1.png" width="1152" /></p>
</div>
<div id="normalization" class="section level2">
<h2>Normalization</h2>
<p>often the technique of rarefying is used to normalize sequencing data, for a variety of reason. Though this method results is throwing out a large amount of real-data to save us from a small amount of false-data. The effectiveness of rarefying is shown in rarefactions curves as seen here.</p>
<pre class="r"><code>abundances &lt;- as.matrix(data.frame(otu_table(phylo.data)))
metadata &lt;- sample_data(phylo.data)
rarecurve(abundances, step = 20, sample = min(rowSds(abundances)), xlab = &quot;Sample Size&quot;, ylab = &quot;Species&quot;, label = TRUE)</code></pre>
<p><img src="bacteria_differential_abundance_analysis_files/figure-html/normalization-1.png" width="1152" /></p>
<pre class="r"><code># rarecurve(abundances, step = 20, sample = 5, xlab = &quot;Sample Size&quot;, ylab = &quot;Species&quot;, label = TRUE)</code></pre>
<p>Alternatively the data can be analyzed using relative abundance, which normalizes each sample but setting the values to 1 and representing each OTU as a proportion of the whole. This retains all the data and allows the samples to be fairly compared with the assumption that all the data is true.</p>
<pre class="r"><code>relative_abundance &lt;- function(physeq, n) {
  physeq.scale &lt;- transform_sample_counts(physeq, function(x){(n * x/sum(x))})
  otu_table(physeq.scale) &lt;- floor(otu_table(physeq.scale))
  physeq.scale &lt;- prune_taxa(taxa_sums(physeq.scale) &gt; 0, physeq.scale)
  return(physeq.scale)
}

relabundance &lt;- relative_abundance(tax_glom(phylo.data, taxrank = &quot;phylum&quot;), 100)
    relabundance_table &lt;- psmelt(relabundance)
    relabundance_table &lt;- relabundance_table[relabundance_table$Abundance &gt; 0.01,]
    relabundance_table &lt;- subset(relabundance_table, select=-c(domain,sample.name))

p &lt;- ggplot(relabundance_table, aes(x = as.factor(Sample), y = Abundance, fill = phylum))
    p &lt;- p + geom_bar(stat = &quot;identity&quot;, width = 0.5)
    p &lt;- p + facet_grid(. ~ Treatment, space = &quot;free&quot;, scales = &quot;free&quot;)
    p &lt;- p + theme_bw()
    p &lt;- p + theme(axis.title.x=element_blank(),
                    axis.text.x=element_blank(),
                    axis.ticks.x=element_blank())
    p &lt;- p + theme(legend.key.size = unit(0.5,&quot;line&quot;))
    p &lt;- p + guides(fill=guide_legend(ncol=1))
    p &lt;- p + ylab(&quot;Relative Abundance (Phylum &gt; 1%) \n&quot;)
p</code></pre>
<p><img src="bacteria_differential_abundance_analysis_files/figure-html/normalization-2-1.png" width="1152" /></p>
</div>
<div id="ordination" class="section level2">
<h2>Ordination</h2>
<p>MDS plots show us the variation between samples and when looked at by treaments, shows us how much variation exists between each. The circles attempt to show a cluster for the center of the treatment, and we can see how they differentiate, though not to a great extent.</p>
<pre class="r"><code>source(&quot;scripts/ggplot.NMDS.ellipse.R&quot;)
getPalette = colorRampPalette(brewer.pal(8, &quot;Dark2&quot;)); colorCount = length(unique(metadata$Treatment)); colors = getPalette(colorCount); theme_set(theme_bw())
totu&lt;-t(abundances)
mds &lt;- metaMDS(totu, autotransform = F, k = 2, trymax = 100)</code></pre>
<pre class="r"><code>ggplot.NMDS.ellipse(mds, metadata$Treatment, colors)</code></pre>
<p><img src="bacteria_differential_abundance_analysis_files/figure-html/ordinaiton-plot-1.png" width="1152" /></p>
</div>
<div id="analysis-of-variance" class="section level2">
<h2>Analysis of variance</h2>
<p>Adonis is a method commonly used for ecological systems. It runs permutations of subsets of the data and calculates distances from the cnetroid using in the same manner as ANOVA. And thusly we find that none of the treatments actually show statistically significant effects on the bacterial community.</p>
<pre class="r"><code>rela.dist &lt;- phyloseq::distance(relabundance, &quot;bray&quot;)
si &lt;- data.frame(sample_data(relabundance))
pairs &lt;- t(combn(unique(si$Treatment), 2))
df &lt;- data.frame()         
for (i in 1:nrow(pairs)){
    temp.rowname &lt;- paste(pairs[i, 1], pairs[i, 2], sep=&quot;::&quot;)
    temp.phy &lt;- subset_samples(relabundance, Treatment %in% pairs[i, ])
    temp.phy &lt;- prune_taxa(taxa_sums(temp.phy) &gt; 0, temp.phy)
    temp.dist &lt;- phyloseq::distance(temp.phy, &quot;bray&quot;)
    temp.result &lt;- adonis(temp.dist ~ Treatment +  Month + as.factor(Flower), perm=9999, as(sample_data(temp.phy), &quot;data.frame&quot;))
    temp.df &lt;- data.frame(temp.rowname, temp.result$aov.tab[4][1, ], temp.result$aov.tab[5][1, ], temp.result$aov.tab[6][1, ])
    df &lt;- rbind(df, temp.df)
}
names(df)&lt;-c(&quot;factor&quot;, &quot;rela.F.model&quot;, &quot;rela.adonis_R2&quot;, &quot;rela.Pr(&gt;F)&quot;)
df</code></pre>
<pre><code>##            factor rela.F.model rela.adonis_R2 rela.Pr(&gt;F)
## 1 Bagged::Exposed            0    0.011749638           1
## 2   Bagged::Caged            0    0.002311412           1
## 3  Exposed::Caged            0    0.009757328           1</code></pre>
<pre class="r"><code>anova(lm(psmelt(relabundance)$Abundance ~ psmelt(relabundance)$Treatment))</code></pre>
<pre><code>## Analysis of Variance Table
## 
## Response: psmelt(relabundance)$Abundance
##                                  Df  Sum Sq Mean Sq F value Pr(&gt;F)
## psmelt(relabundance)$Treatment    2       0   0.045   2e-04 0.9998
## Residuals                      4729 1009139 213.394</code></pre>
<pre class="r"><code>anova(lm(psmelt(relabundance)$Abundance ~ psmelt(relabundance)$Month))</code></pre>
<pre><code>## Analysis of Variance Table
## 
## Response: psmelt(relabundance)$Abundance
##                              Df  Sum Sq Mean Sq F value Pr(&gt;F)
## psmelt(relabundance)$Month    2       0   0.024   1e-04 0.9999
## Residuals                  4729 1009139 213.394</code></pre>
<pre class="r"><code>anova(lm(psmelt(relabundance)$Abundance ~ psmelt(relabundance)$Flower))</code></pre>
<pre><code>## Analysis of Variance Table
## 
## Response: psmelt(relabundance)$Abundance
##                               Df  Sum Sq Mean Sq F value Pr(&gt;F)
## psmelt(relabundance)$Flower  181       9    0.05   2e-04      1
## Residuals                   4550 1009130  221.79</code></pre>
<pre class="r"><code>TukeyHSD(aov(psmelt(relabundance)$Abundance ~ psmelt(relabundance)$Treatment))</code></pre>
<pre><code>##   Tukey multiple comparisons of means
##     95% family-wise confidence level
## 
## Fit: aov(formula = psmelt(relabundance)$Abundance ~ psmelt(relabundance)$Treatment)
## 
## $`psmelt(relabundance)$Treatment`
##                         diff       lwr      upr     p adj
## Caged-Bagged    0.0107624831 -1.374595 1.396120 0.9998171
## Exposed-Bagged  0.0004683655 -1.149380 1.150317 0.9999995
## Exposed-Caged  -0.0102941176 -1.298125 1.277537 0.9998064</code></pre>
</div>
<div id="differential-abundance" class="section level2">
<h2>Differential Abundance</h2>
<p>There are several different packages and methods for looking at differential abundance (taken from differential expresson used in gene studies). Here we used DESeq2 in R, and found bothing interesting when comparing each treatment to one another.</p>
<pre class="r"><code>abundance &lt;- relative_abundance(tax_glom(phylo.data, taxrank = &quot;class&quot;), 100)
caged &lt;- subset_samples(abundance, Treatment!=&quot;Bagged&quot;)
caged &lt;- prune_taxa(taxa_sums(caged) &gt; 0, caged)
# caged &lt;- transform_sample_counts(caged, function(x){x/sum(x)})
caged.ds &lt;- phyloseq_to_deseq2(caged, ~ Treatment)
bagged &lt;- subset_samples(abundance, Treatment!=&quot;Caged&quot;)
bagged &lt;- prune_taxa(taxa_sums(bagged) &gt; 0, bagged)
# bagged &lt;- transform_sample_counts(bagged, function(x){x/sum(x)})
bagged.ds &lt;- phyloseq_to_deseq2(bagged, ~ Treatment)</code></pre>
<pre class="r"><code>caged.res &lt;- data.frame(results(DESeq(caged.ds, test=&quot;Wald&quot;, fitType=&quot;parametric&quot;), cooksCutoff=F))
bagged.res &lt;- data.frame(results(DESeq(bagged.ds, test=&quot;Wald&quot;, fitType=&quot;parametric&quot;), cooksCutoff=F))

caged.res.sig &lt;- subset(caged.res, padj &lt; 0.1) 
bagged.res.sig &lt;- subset(bagged.res, padj &lt; 0.1) 

caged.res.sig &lt;- merge(caged.res.sig, tax_table(phylo.data), by = &quot;row.names&quot;)
bagged.res.sig &lt;- merge(bagged.res.sig, tax_table(phylo.data), by = &quot;row.names&quot;)</code></pre>
<pre class="r"><code>caged.res.sig$pos_neg &lt;- ifelse(caged.res.sig$log2FoldChange &gt; 0, &quot;positive&quot;, &quot;negative&quot;)
caged.res.sig$treatment &lt;- &quot;caged&quot;
bagged.res.sig$pos_neg &lt;- ifelse(bagged.res.sig$log2FoldChange &gt; 0, &quot;positive&quot;, &quot;negative&quot;)
#bagged.res.sig$treatment &lt;- &quot;bagged&quot;
sig.otus &lt;- rbind(caged.res.sig, bagged.res.sig)

p &lt;- ggplot(data=sig.otus, aes(x=class, y=log2FoldChange))
  p &lt;- p + geom_point(aes(color=treatment))
  p &lt;- p + theme_bw()
  p &lt;- p + theme(axis.text.x=element_text(angle=90, hjust = 1, vjust = 0.5),
                legend.position = &quot;top&quot;,
                axis.text.y=element_text(size =12),
                axis.title.x = element_text(face=&quot;bold&quot;, size =14), 
                axis.title.y = element_text(face=&quot;bold&quot;, size = 14)
            )
  p &lt;- p + geom_hline(yintercept=0)
  p &lt;- p + scale_color_brewer(palette=&quot;Dark2&quot;)
  p &lt;- p + xlab(&quot;Class&quot;) + ylab(&quot;Differences in Abundance (log2)&quot;)
  p &lt;- p + theme(axis.title.x = element_blank())
p</code></pre>
<p><img src="bacteria_differential_abundance_analysis_files/figure-html/DESeq-plot-1.png" width="1152" /></p>
</div>
<div id="beta-diversity" class="section level2">
<h2>Beta-Diversity</h2>
<p>Using the <span class="math inline">\(\beta\)</span>-diversity function published in Vannette et al, We caclulated the <span class="math inline">\(\beta\)</span>-diversity for each treatment to try to replicate their results. We saw an increase in <span class="math inline">\(\beta\)</span>-diversity for the caged and exposed from the bagged, but not as drastic as in the original study.</p>
<pre class="r"><code>source(&quot;scripts/mds.envfit.arrows.R&quot;);source(&quot;scripts/ordi.sf.R&quot;);source(&quot;scripts/Vannette.R&quot;);source(&quot;scripts/Tello.R&quot;);

caged &lt;- subset_samples(abundance, Treatment==&quot;Caged&quot;)
caged &lt;- prune_taxa(taxa_sums(caged) &gt; 0, caged)
bagged &lt;- subset_samples(abundance, Treatment==&quot;Bagged&quot;)
bagged &lt;- prune_taxa(taxa_sums(bagged) &gt; 0, bagged)
exposed &lt;- subset_samples(abundance, Treatment==&quot;Exposed&quot;)
exposed &lt;- prune_taxa(taxa_sums(exposed) &gt; 0, exposed)</code></pre>
<pre class="r"><code># bses_full &lt;- beta.ses.list(phylo.data)
# names(bses_full) &lt;- rownames(sample_data(phylo.data))
bses_bagged &lt;- beta.ses.list(bagged)
names(bses_bagged) &lt;- rownames(sample_data(bagged))
bses_caged &lt;- beta.ses.list(caged)
names(bses_caged) &lt;- rownames(sample_data(caged))
bses_exposed &lt;- beta.ses.list(exposed)
names(bses_exposed) &lt;- rownames(sample_data(exposed))

betas &lt;- list(bses_bagged,bses_caged,bses_exposed)
names(betas) &lt;- c(&quot;bagged&quot;,&quot;caged&quot;,&quot;exposed&quot;)
saveRDS(betas, &quot;data/bacteria/germs_otus/beta_diversities.RDS&quot;)
betas &lt;- readRDS(&quot;data/bacteria/germs_otus/beta_diversities.RDS&quot;)

bses_bagged &lt;- betas$bagged
bses_caged &lt;- betas$caged
bses_exposed &lt;- betas$exposed</code></pre>
<pre class="r"><code>getPalette = colorRampPalette(brewer.pal(8, &quot;Dark2&quot;))
colorCount = length(unique(sample_data(phylo.data)$Treatment))
colors = getPalette(colorCount)

datta &lt;- melt(c(data.frame(bagged = bses_bagged), data.frame(caged = bses_caged), data.frame(exposed = bses_exposed))); colnames(datta) &lt;- c(&#39;Beta&#39;, &#39;Treatment&#39;)

ggplot(datta, aes(x = Treatment, y = Beta)) +
  geom_boxplot(colour = colors)</code></pre>
<p><img src="bacteria_differential_abundance_analysis_files/figure-html/plot-beta-1.png" width="1152" /></p>
<p>So maybe the rarefying process had a significant affect on the data?</p>
<pre class="r"><code>rare_data &lt;- rarefy_even_depth(phylo.data, sample.size = 500, replace = FALSE)
rare_data &lt;- relative_abundance(rare_data, 100)</code></pre>
</div>
<div id="beta-diversity-of-rarefied-samples" class="section level2">
<h2>Beta-Diversity of rarefied samples</h2>
<pre class="r"><code>caged &lt;- subset_samples(rare_data, Treatment==&quot;Caged&quot;)
caged &lt;- prune_taxa(taxa_sums(caged) &gt; 0, caged)
bagged &lt;- subset_samples(rare_data, Treatment==&quot;Bagged&quot;)
bagged &lt;- prune_taxa(taxa_sums(bagged) &gt; 0, bagged)
exposed &lt;- subset_samples(rare_data, Treatment==&quot;Exposed&quot;)
exposed &lt;- prune_taxa(taxa_sums(exposed) &gt; 0, exposed)</code></pre>
<pre class="r"><code># bses_full &lt;- beta.ses.list(phylo.data)
# names(bses_full) &lt;- rownames(sample_data(phylo.data))
bses_bagged &lt;- beta.ses.list(bagged)
names(bses_bagged) &lt;- rownames(sample_data(bagged))
bses_caged &lt;- beta.ses.list(caged)
names(bses_caged) &lt;- rownames(sample_data(caged))
bses_exposed &lt;- beta.ses.list(exposed)
names(bses_exposed) &lt;- rownames(sample_data(exposed))

betas &lt;- list(bses_bagged,bses_caged,bses_exposed)
names(betas) &lt;- c(&quot;bagged&quot;,&quot;caged&quot;,&quot;exposed&quot;)
saveRDS(betas, &quot;data/bacteria/germs_otus/beta_diversities.RDS&quot;)
betas &lt;- readRDS(&quot;data/bacteria/germs_otus/beta_diversities.RDS&quot;)

bses_bagged &lt;- betas$bagged
bses_caged &lt;- betas$caged
bses_exposed &lt;- betas$exposed</code></pre>
<pre class="r"><code>getPalette = colorRampPalette(brewer.pal(8, &quot;Dark2&quot;))
colorCount = length(unique(sample_data(phylo.data)$Treatment))
colors = getPalette(colorCount)

datta &lt;- melt(c(data.frame(bagged = bses_bagged), data.frame(caged = bses_caged), data.frame(exposed = bses_exposed))); colnames(datta) &lt;- c(&#39;Beta&#39;, &#39;Treatment&#39;)

ggplot(datta, aes(x = Treatment, y = Beta)) +
  geom_boxplot(colour = colors)</code></pre>
<p><img src="bacteria_differential_abundance_analysis_files/figure-html/rare-plot-beta-1.png" width="1152" /></p>
<pre class="r"><code>rare_data &lt;- rarefy_even_depth(phylo.data, sample.size = 500, replace = FALSE)
abundance &lt;- relative_abundance(tax_glom(rare_data, taxrank = &quot;class&quot;), 100)
caged &lt;- subset_samples(abundance, Treatment!=&quot;Bagged&quot;)
caged &lt;- prune_taxa(taxa_sums(caged) &gt; 0, caged)
# caged &lt;- transform_sample_counts(caged, function(x){x/sum(x)})
caged.ds &lt;- phyloseq_to_deseq2(caged, ~ Treatment)
bagged &lt;- subset_samples(abundance, Treatment!=&quot;Caged&quot;)
bagged &lt;- prune_taxa(taxa_sums(bagged) &gt; 0, bagged)
# bagged &lt;- transform_sample_counts(bagged, function(x){x/sum(x)})
bagged.ds &lt;- phyloseq_to_deseq2(bagged, ~ Treatment)</code></pre>
<pre class="r"><code>caged.res &lt;- data.frame(results(DESeq(caged.ds, test=&quot;Wald&quot;, fitType=&quot;parametric&quot;), cooksCutoff=F))
bagged.res &lt;- data.frame(results(DESeq(bagged.ds, test=&quot;Wald&quot;, fitType=&quot;parametric&quot;), cooksCutoff=F))

caged.res.sig &lt;- subset(caged.res, padj &lt; 0.1) 
bagged.res.sig &lt;- subset(bagged.res, padj &lt; 0.1) 

caged.res.sig &lt;- merge(caged.res.sig, tax_table(phylo.data), by = &quot;row.names&quot;)
bagged.res.sig &lt;- merge(bagged.res.sig, tax_table(phylo.data), by = &quot;row.names&quot;)</code></pre>
<pre class="r"><code>caged.res.sig$pos_neg &lt;- ifelse(caged.res.sig$log2FoldChange &gt; 0, &quot;positive&quot;, &quot;negative&quot;)
caged.res.sig$treatment &lt;- &quot;caged&quot;
bagged.res.sig$pos_neg &lt;- ifelse(bagged.res.sig$log2FoldChange &gt; 0, &quot;positive&quot;, &quot;negative&quot;)
#bagged.res.sig$treatment &lt;- &quot;bagged&quot;
sig.otus &lt;- rbind(caged.res.sig, bagged.res.sig)

p &lt;- ggplot(data=sig.otus, aes(x=class, y=log2FoldChange))
  p &lt;- p + geom_point(aes(color=treatment))
  p &lt;- p + theme_bw()
  p &lt;- p + theme(axis.text.x=element_text(angle=90, hjust = 1, vjust = 0.5),
                legend.position = &quot;top&quot;,
                axis.text.y=element_text(size =12),
                axis.title.x = element_text(face=&quot;bold&quot;, size =14), 
                axis.title.y = element_text(face=&quot;bold&quot;, size = 14)
            )
  p &lt;- p + geom_hline(yintercept=0)
  p &lt;- p + scale_color_brewer(palette=&quot;Dark2&quot;)
  p &lt;- p + xlab(&quot;Class&quot;) + ylab(&quot;Differences in Abundance (log2)&quot;)
  p &lt;- p + theme(axis.title.x = element_blank())
p</code></pre>
<p><img src="bacteria_differential_abundance_analysis_files/figure-html/rare-DESeq-plot-1.png" width="1152" /></p>
</div>

<p><br>
<br>

<strong><a href="https://github.com/pommevilla">Paul Villanueva</a></strong> and 
<strong><a href="https://github.com/sdsmith1390">Schuyler Smith</a></strong>
<br>
Ph.D. Students - Bioinformatics and Computational Biology<br>
Iowa State University.  Ames, IA.</p>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
